### PR TITLE
Feature/init av ai branch at feature c41a5b93

### PR DESCRIPTION
**Pull request by metelik**

### CODE REVIEW

**Review by adrigonzo — APPROVED** [review]

File: src/bacnet/bacdef.h [hunk 1]
```diff
@@ -187,6 +195,24 @@ typedef struct BACnet_Object_Id {
     uint32_t instance;
 } BACNET_OBJECT_ID;
 
+/* This Struct is for initialisation info coming from Elixir 
+*/
+typedef struct BACnet_Object_Init_s {
+    // Instance works as an index to the object within the type, the type number is not included in this value.
+    uint32_t Object_Instance;
+    //BACNET_CHARACTER_STRING Object_Name;
```
Comment (metelik) [inline src/bacnet/bacdef.h:203]: Why? We could properly initialise the variable BACNET_CHARACTER_STRING type calling the appropriate API from bacnet NIF.

File: src/bacnet/basic/object/av.c [hunk 2]
```diff
@@ -128,18 +128,68 @@ void Analog_Value_Init(void)
...
+    return false;
+  }
+
+  if (pInit_data->length > MAX_ANALOG_VALUES) {
+    return false;
+  }
+
+  for (i = 0; i < pInit_data->length; i++) {
+    if (pInit_data->Object_Init_Values[i].Object_Instance < BACNET_MAX_INSTANCE) {
+      AV_Descr[i].Object_Instance = pInit_data->Object_Init_Values[i].Object_Instance;
```
Comment (metelik) [inline src/bacnet/basic/object/av.c:151]: I was wondering... Why not to create an API function i.e. Analog_Value_Instance_Set(index, instance_no), export it via NIF and do the looping from Elixir side? Other funcs like Name_Set and Description_Set are already present....

Comment (adrigonzo) [reply src/bacnet/basic/object/av.c:151]: NIF calls are expensive, trying to pass in the data you need to the other side, and handle it there seemed better

File: src/bacnet/bacdef.h (hunk 1 above)
Comment (adrigonzo) [reply src/bacnet/bacdef.h:203]: it was just an extra layer of complication when all `BACNET_OBJECT_INIT_T` needed to do was match the passed in `{object_name, object_index, description, unit}`. It keeps that part of the init process simpler, without having to init the fields within `BACNET_CHARACTER_STRING`.
However, if we do need to support UTF8 then we'll need to handle that as well in the code, as currently we assume it's always simple ASCII. I haven't played around with charlist and binary conversion through the NIF recently.

**Pull request merged**